src/data/prayers/data-source.js

```diff
@@ -304,15 +304,15 @@ export default class Prayer extends RockApolloDataSource {
     }
   };
 
-  answer = async (id, answer) => {
+  editPrayer = async (prayer) => {
```

Comment:
destructure this guy and set defaults. that's gonna throw an error if anything is undefined.

Reply:
Actually, if I default `answer = null` here then I can keep from having to set it in the client if I wanted to when I'm removing an answer. I've been setting it to `null` because that's what `answer` is on prayers that don't have one out of the box.

Reply:
so how are you suggesting the query will work to remove a prayer?

Reply:
It can _technically_ work 2 ways. I can pass `{id: ###, answer: null}` (which is probably more right because it spells out my intentions or I can just pass `{ id: ### }` and it will auto-set the answer to `null` by default. Either way will remove the prayer answer.

Reply:
so if we extend this to later be able to edit the prayer text, you would do `{id: ###, text: "new text"}` which would wipe out the answer. `undefined` in my mind means _do nothing_. an empty string or null says _overwrite_.

Reply:
Yeah, I see what you're saying there. So then does the top half of that function become something like this? Not sure how to conditionally set the pieces of the object without doing it all before we run the `patch`. Right now it's just one `if` but if we kept adding things it could get kind of crazy.
```
editPrayer = async ({ id, answer = undefined }) => {
    const thisObject = {};
    if (answer !== undefined) {
      thisObject.Answer = answer;
    }
    try {
      await this.patch(`/PrayerRequests/${id}`, thisObject);
```

Reply:
So basically I end up with something like:
```
  editPrayer = async ({ id, answer = undefined }) => {
    const prayer = {};
    if (answer !== undefined) {
      prayer.Answer = answer;
    }
    try {
      if (!isEmpty(prayer)) {
        await this.patch(`/PrayerRequests/${id}`, prayer);
      }
      return this.getFromId(id);
    } catch (e) {
      bugsnagClient.notify(new Error('Editing prayer answer failed.'), {
        metaData: { rockPrayerID: id },
        severity: 'warning',
      });
      return null;
    }
  };
```

Reply:
I found a _sort of_ cleaner (if not a bit harder to understand) way of setting the pieces of the object without needing the multiple `ifs`. 

```
const prayer = {
      ...(answer !== undefined && { Answer: answer }),
    };
```